config/static/nodes/cri-o/install-custom-cri-o.yaml.envsubst

```diff
@@ -0,0 +1,30 @@
+---
+# https://github.com/openshift/machine-config-operator/tree/master/docs
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: 99-install-custom-cri-o-${ROLE}
+  labels:
+    machineconfiguration.openshift.io/role: ${ROLE}
+spec:
+  config:
+    ignition:
+      version: "3.2.0"
+    systemd:
+      units:
+      - name: "install-cri-o.service"
+        enabled: true
+        # For getting the logs run:
+        # $ oc adm node-logs node/NODE -u install-cri-o
+        contents: |
+          [Unit]
+          Description=Install custom cri-o
+          # https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/
+          After=network-online.target
+
+          [Service]
+          Type=oneshot
+          ExecStart=bash -c 'rpm -q "$( rpm -q ${RPM_CRI_O} )" || rpm-ostree override replace --reboot "${RPM_CRI_O}"'
```

Comment:
Side note: we are usually overriding `runc`, not `cri-o` :)

From this command construction, I assume that we were unable to get it to work with only
`rpm-ostree override replace --idempotent --reboot`.  That is, was it unconditionally rebooting?
And is that why we must explicitly test the package version via `rpm` program?

I don't like that the package must be unconditionally downloaded upon every reboot, in order to perform
this test.  Perhaps two separate variables are better - one for the package name and the other for the
URL?  The `Makefile` can even take care of this, deriving the package name from the RPM and setting a
variable - rather than this being done by systemd on the worker node, every time it boots.

Reply:
about cri-o: Yea, I know! but the effort was a little to get the config for cri-o too, so we have the automation just in case.
about the command: I was using previously`'rpm-ostree install --idempotent --reboot`, but I should have tested wrongly; when I was doing final tests I saw that it was not installing the package and I searched for a workaround.

Thanks @frasertweedale for point out the uncondicional download, I will update the PR for using two variables, so it is downloaded only when it is needed.

Reply:
I think the new approach is good, avoiding the unconditional download during bootup and only downloading it if the desired RPM is not installed. :+1: